### PR TITLE
Add context (res id and name) when downloading data

### PR DESCRIPTION
**Pull request by Moohan**
The problem was highlighted in [this teams chat](https://teams.microsoft.com/l/message/19:9620ef6cf8234d50a0f95caba65a3edf@thread.tacv2/1711384080545?tenantId=10efe0bd-a030-4bca-809c-b5e6745e499a&groupId=ec4250f9-b70a-4f32-9372-a232ccb4f713&parentMessageId=1711384080545&teamName=PHS%20Data%20and%20Intelligence%20Forum&channelName=Posit%20Technical%20Queries&createdTime=1711384080545), when using `get_dataset` if the dataset doesn't include a variable like `date` it becomes impossible to tell which rows are from which resource as `get_dataset` just joins them into a single tibble.

This PR introduces an `include_context` parameter which is `FALSE` by default to both `get_resource` and `get_dataset` if `TRUE` they will now append a few variables: `"res_id", "res_name", "res_created_date" and "res_modified_date"` to the **front** of the data.

For `get_dataset` this is achieved with no additional API calls as all the needed info is included when we hit `package_show` for the list of IDs. For `get_resource` this involves an additional API call to `resource_show` (we haven't used this before).

I chose to have them both be `FALSE` by default so it won't break any existing code.

### CODE REVIEW

**Comment by csillasch** [comment]
This is great! I`m probably leaning toward having it optional (but perhaps default to TRUE for get_dataset) and I think it might be worth having the option for get_resource as well (but would perhaps default to FALSE for get_resource)?

**Comment by Moohan** [comment]
From discussion

- [ ] Add some more tests with other datasets and resources (d3eaaf84-0f3b-4fb8-9460-e33503095fbe).
- [ ] Should missing modified date be created date.
- [ ]

**Comment by ross-hull** [comment]
This looks good to me. And using it as a user was simple. One thing that I wondered is if it is worth having the same naming convention between the new context columns, as is already used in the open data? The context columns are snake case while the others are upper camel.

**Review by csillasch — CHANGES_REQUESTED** [review]
Looks good! 

I spotted unexpected behaviour when testing on the General Practitioner Contact Details dataset: modified date is returned correctly using get_dataset but returns NULL with get_resource("647b256e-4a03-4963-8402-bf559c9e2fff")

Suggested a change to sort this which works for the resource but might be worth testing against some of the others - especially where it worked fine previously.

**Comment by Moohan** [comment]
> This looks good to me. And using it as a user was simple. One thing that I wondered is if it is worth having the same naming convention between the new context columns, as is already used in the open data? The context columns are snake case while the others are upper camel.

Fair enough, do these names look good?
```
"ResId" (or "ResID"?)
"ResName"
"ResCreatedDate"
"ResModifiedDate"
```

File: R/get_resource.R [hunk 1]
```diff
@@ -35,51 +51,71 @@ get_resource <- function(res_id, rows = NULL, row_filters = NULL, col_select = N
...
+    # Get resource context if required
+    context_content <- phs_GET(
+      action = "resource_show",
+      query = paste0("id=", res_id)
+    )
+
+    res_id <- context_content$result$id
+    res_name <- context_content$result$name
+    res_created_date <- context_content$result$created
+    res_modified_date <- context_content$result$modified
```
[the remark this answers is not in the document]
Comment (Moohan) [reply R/get_resource.R:109]: I started writing a test for this (your suggestion does seem to fix) but I noticed another issue where the date type changes when binding together the dataset.

https://github.com/r-lib/vctrs/issues/1930

**Review by csillasch — COMMENTED** [review]
Get_dataset seems to handle it fine but get_resource is returning the dates as lists - all your changes look OK to me so not sure how/why this is happening

**Review by csillasch — APPROVED** [review]

**Pull request merged**